### PR TITLE
[Merged by Bors] - feat(CategoryTheory): define the notion of effectively enough objects with respect to a functor

### PR DESCRIPTION
**Pull request by dagurtomas**
The notion means that for every object in the target, there is an effective epi to it from an object in the image of the functor.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by joelriou** [comment]
Could you add a TODO section in the file where you would give examples of situations where this new notion shall be relevant? Otherwise, it seems ok to me.

**Comment by dagurtomas** [comment]
I added some TODOs, which will be completed in #11518.

File: Mathlib/CategoryTheory/EffectiveEpi/Enough.lean [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+epimorphism `F.π : F.over X ⟶ X`.
+-/
+noncomputable def over (X : D) : D :=
+  F.obj (EffectivelyEnough.presentation (F := F) X).some.p
+
+/--
+The epimorphism `F.π : F.over X ⟶ X` from the arbitrarily chosen object in the image of `F`
+over `X`.
+-/
+noncomputable def π (X : D) : over F X ⟶ X :=
```
Comment (joelriou) [inline Mathlib/CategoryTheory/EffectiveEpi/Enough.lean:61]: This definition (and that of `over`) should either be in a more specialized namespace, or the names should be made slightly longer.

File: Mathlib/CategoryTheory/EffectiveEpi/Enough.lean [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+epimorphism `F.morphismOver : F.objOver X ⟶ X`.
+-/
+noncomputable def objOver (X : D) : D :=
+  F.obj (EffectivelyEnough.presentation (F := F) X).some.p
+
+/--
+The epimorphism `F.morphismOver : F.objOver X ⟶ X` from the arbitrarily chosen object in the image
+of `F` over `X`.
+-/
+noncomputable def morphismOver (X : D) : F.objOver X ⟶ X :=
```
Comment (joelriou) [inline Mathlib/CategoryTheory/EffectiveEpi/Enough.lean:61]: I think the following name would be more specific:
```suggestion
noncomputable def effectiveEpiOver (X : D) : F.objOver X ⟶ X :=
```
Then, `objOver` could be renamed `effectiveEpiOverObj`.

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**